### PR TITLE
.github/workflows: Avoid running the e2e-tests workflow on doc-only PRs

### PR DESCRIPTION
**Pull request by timflannagan**
Update the e2e-tests.yml github workflow and avoid running that action
on PRs that only change the doc directory.

**Description of the change:**

**Motivation for the change:**

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/doc`
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

File: .github/workflows/e2e-tests.yml [hunk 1]
```diff
@@ -4,6 +4,9 @@ on:
     branches:
       - master
   pull_request:
+    paths:
+      - '**'
+      - '!doc/**'
```
Comment (joelanford) [inline .github/workflows/e2e-tests.yml:9]: Nit: also ignore markdown files? 
```suggestion
      - '!doc/**'
      - '!**/*.md'
```

**Review by joelanford — COMMENTED** [review]
Just a non-blocking nit.

/lgtm

File: .github/workflows/e2e-tests.yml (hunk 1 above)
Comment (timflannagan) [reply .github/workflows/e2e-tests.yml:9]: Yeah that's not a bad idea - any scenario where we would only want to restrict that to _root_ markdown files? Or extend this to changes made to similar files, like `.gitignore`? I'm mainly following suit w.r.t to the other workflows that disable that doc/ directory path, so maybe that's worth tackling down the line.

**Pull request merged**